### ISSUE TITLE
OIDC signing up flow crashes when a required field isn't provided

### ISSUE THREAD

**Issue by renom** [ory/kratos#2660]
### Preflight checklist

- [X] I could not find a solution in the existing issues, docs, nor discussions.
- [X] I agree to follow this project's [Code of Conduct](https://github.com/ory/kratos/blob/master/CODE_OF_CONDUCT.md).
- [X] I have read and am following this repository's [Contribution Guidelines](https://github.com/ory/kratos/blob/master/CONTRIBUTING.md).
- [ ] This issue affects my [Ory Cloud](https://www.ory.sh/) project.
- [ ] I have joined the [Ory Community Slack](https://slack.ory.sh).
- [ ] I am signed up to the [Ory Security Patch Newsletter](https://ory.us10.list-manage.com/subscribe?u=ffb1a878e4ec6c0ed312a3480&id=f605a41b53).

### Describe the bug

When there's a field in the schema which is required, but the value that is obtained from the claims is actually empty (so the field isn't set after applying the jsonnet file), the flow crashes with a schema validation error (the error message is `missing properties: "property_name"`). Not sure whether the issue is relevant to kratos or kratos-selfservice-ui-node, but since the property isn't provided, kratos-selfservice-ui-node should ask to provide it instead of interrupting the flow.

### Reproducing the bug

Don't provide a required property in the traits, or provide a value of a string property with a length < minLength.

### Relevant log output

```shell
time=2022-08-11T00:00:00Z level=info msg=Encountered self-service flow error. func=github.com/ory/kratos/selfservice/flow/registration.(*ErrorHandler).WriteFlowError file=/project/selfservice/flow/registration/error.go:76 audience=audit error=map[message:I[#/traits] S[#/properties/traits/required] missing properties: "property_name"
```

### Relevant configuration

```yml
kratos with kratos-selfservice-ui-node
```

### Version

v0.10.1

### On which operating system are you observing this issue?

Linux

### In which environment are you deploying?

Docker Compose

**Comment by aeneasr** [comment]
This looks like a JSON Schema misconfiguration, because the error says that the field `property_name` does not exist in your schema.

**Comment by renom** [comment]
> This looks like a JSON Schema misconfiguration, because the error says that the field `property_name` does not exist in your schema.

Well, I changed the jsonnet file to set the field with an empty-string value (`"property_name": ""`) when apple doesn't return a value, now it prints "length must be >= 1, but got 0" in the log. It still interrupts the registration flow, instead of showing the form to provide missing (but required) fields. Also I checked the property name, it exists in the schema.
